### PR TITLE
Add ingredient_config resource.

### PR DESCRIPTION
**Pull request by sersut**
@chef-cookbooks/engineering-services I think this will be a good starting point. We can add file (or template) based config file placement or move to a `block` based evaluation instead of string collection if we need to in the future.

@jtimberman Let me know if you think we should capture the `config-file` information in `PRODUCT_MATRIX.md`.

### CODE REVIEW

**Comment by sersut** [comment]
Fixes #7

File: README.md [hunk 1]
```diff
@@ -73,6 +85,25 @@ omnibus_service 'chef-server-core/rabbitmq' do
 end
 ```
 
+To install Chef Server using some custom configuration options:
+
+```ruby
+chef_ingredient "chef-server" do
+  config <<-EOS
+api_fqdn "#{node["fqdn"]}"
+ip_version "ipv6",
```
Comment (jtimberman) [inline README.md]: stray comma?

File: libraries/helpers.rb [hunk 2]
```diff
@@ -91,20 +91,76 @@ def reconfigure
...
+        },
+        'chefdk' => {
+          'package-name' => 'chefdk',
+          'ctl-command'  => nil,
+          'config-file'  => nil
+        },
+        'delivery' => {
+          'package-name' => 'delivery',
+          'ctl-command'  => 'delivery-ctl',
+          'config-file'  => nil
```
Comment (jtimberman) [inline libraries/helpers.rb]: Doesn't delivery have a default config file?

I see an `/etc/delivery/delivery.rb` on our internal delivery server.

File: test/fixtures/cookbooks/test/recipes/config_install.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  action :install
+end
+
+ingredient_config "manage" do
+  notifies :reconfigure, "chef_ingredient[manage]"
+end
+
+ingredient_config "chef-server" do
+  notifies :reconfigure, "chef_ingredient[chef-server]"
+end
```
Comment (schisamo) [inline test/fixtures/cookbooks/test/recipes/config_install.rb]: We'll want to add a Test Kitchen suite and some ServerSpec integration test coverage for this fixture recipe.

**Comment by schisamo** [comment]
@sersut Modulo my integration test coverage comment I'm :+1:.

File: libraries/helpers.rb [hunk 4]
```diff
@@ -91,20 +91,76 @@ def reconfigure
...
+        },
+        'push-server' => {
+          'package-name' => 'chef-push-server',
+          'ctl-command'  => 'chef-push-ctl',
+          'config-file'  => '/etc/opscode-push-jobs-server/opscode-push-jobs-server.rb'
+        },
+        'reporting' => {
+          'package-name' => 'opscode-reporting',
+          'ctl-command'  => 'opscode-reporting-ctl',
+          'config-file'  => nil
```
Comment (jtimberman) [inline libraries/helpers.rb]: config file for reporting is `/etc/opscode-reporting/opscode-reporting.rb`.

File: spec/unit/recipes/test_config_install_spec.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        node.set['chef_admin'] = "admin@chef.io"
+      end.converge(described_recipe)
+    end
+
+    it "creates config directory" do
+      expect(chef_run).to create_directory("/etc/opscode")
+      expect(chef_run).to create_directory("/etc/opscode-manage")
+      expect(chef_run).to create_file("/etc/opscode/chef-server.rb").with content: <<-EOS
+api_fqdn "fauxhai.local"
+ip_version "ipv6",
```
Comment (jtimberman) [inline spec/unit/recipes/test_config_install_spec.rb]: stray comma?

**Comment by jtimberman** [comment]
Couple additional comments:

I don't think we need to add the config info to the PRODUCT_MATRIX.md. We don't have the ctl commands in there, for example. Users can look at the code for the full thing.

You should run rubocop -a :smile:.

What about putting the ingredient's config stuff in the repo recipe and testing in its, rather than making another test suite/recipe? The test kitchen run on the entire list of instances takes at least an hour.

File: README.md (hunk 1 above)
Comment (sersut) [inline README.md]: :+1:

File: libraries/helpers.rb (hunk 2 above)
Comment (sersut) [inline libraries/helpers.rb]: :+1:

File: libraries/helpers.rb (hunk 4 above)
Comment (sersut) [inline libraries/helpers.rb]: :+1:

File: spec/unit/recipes/test_config_install_spec.rb (hunk 5 above)
Comment (sersut) [inline spec/unit/recipes/test_config_install_spec.rb]: :+1:

**Comment by sersut** [comment]
@schisamo / @jtimberman thanks for the feedback and letting me know of rubocop setup :smile: 

I think adding the config stuff to repo recipe is a good idea. I'll make that change soon too.

**Comment by sersut** [comment]
@jtimberman / @schisamo updates are done. Kitchen tests are looking good. Can you guys check this out?

**Pull request merged**